### PR TITLE
Use BlockId instead of String

### PR DESCRIPTION
**Pull request by tcharding**
When doing property based testing on a hash data field (of type
`String`) quickcheck generates input that causes a panic.  The generated
input is not possible during normal program flow.  To inhibit quickcheck
from generating input that is not useful to us we can use a specific
type instead of a string.  We already have code in tree doing the same
sort of thing, we can do the same here.

Use `bitcoin_support::BlockId` instead of `String`.

Resolves: #1506

### CODE REVIEW

File: btsieve/src/bitcoin/blockchain_info_connector.rs [hunk 1]
```diff
@@ -38,7 +38,8 @@ impl BlockchainInfoConnector {
         })
     }
 
-    fn block_by_hash_url(block_hash: &str) -> Url {
+    fn block_by_hash_url(block_hash: &bitcoin_support::BlockId) -> Url {
+        let block_hash = format!("{}", block_hash);
```
Comment (D4nte) [inline btsieve/src/bitcoin/blockchain_info_connector.rs]:
```suggestion
        let block_hash = block_hash.to_string();
```

**Comment by tcharding** [comment]
> Judging from your PR description, I get the feeling you were 100% aware of what you are doing here, even though it was quite a small change and could have been blindly copied from the old PR! Well done :)

Thanks man, that is one of the reasons for writing descriptive PRs and commit messages IMO - to make sure one _realy_ understands what they are doing.  Unfortunately that implies that you can almost always read lack of information in these places from me as not understanding.

**Comment by thomaseizinger** [comment]
@D4nte I hope you don't mind:

bors r+

**Comment by D4nte** [comment]
> @D4nte I hope you don't mind:
> 
> bors r+

Shouln't the author of the PR be the one to `bors r+`?

**Comment by thomaseizinger** [comment]
> > @D4nte I hope you don't mind:
> > bors r+
> 
> Shouln't the author of the PR be the one to `bors r+`?

Anyone can I guess, but since this one fixes a bug with for a flaky build, I wanted to get it in ASAP.

**Pull request merged**